src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -441,44 +506,323 @@ public int getID()
+                this.triggerAchievement(ModAchievements.achievementCatchOneFish);
+                break;
+            case TAG_FARMER_WHEAT:
+                this.triggerAchievement(ModAchievements.achievementHarvestOneWheat);
+                break;
+            case TAG_FARMER_POTATOES:
+                this.triggerAchievement(ModAchievements.achievementHarvestOnePotato);
+                break;
+            case TAG_FARMER_CARROTS:
+                this.triggerAchievement(ModAchievements.achievementHarvestOneCarrot);
+                break;
+            case TAG_LUMBERJACK_SAPLINGS:
+                this.triggerAchievement(ModAchievements.achievementPlantOneSapling);
+                break;
+            case TAG_LUMBERJACK_TREES:
+                this.triggerAchievement(ModAchievements.achievementFellOneTree);
+                break;
+            default:
+                break;
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_THIRD)
+    }
+    /**
+     * Trigger second achievement.
+     * @param statistic the statistic.
+     */
+    private void triggerSecondAchievement(@NotNull String statistic)
+    {
+        switch (statistic)
         {
-            this.triggerAchievement(ModAchievements.achievementKill100Mobs);
+            case TAG_GUARD_MOBS:
+                this.triggerAchievement(ModAchievements.achievementKill25Mobs);
+                break;
+            case TAG_MINER_ORES:
+                this.triggerAchievement(ModAchievements.achievementMine25Ores);
+                break;
+            case TAG_MINER_DIAMONDS:
+                this.triggerAchievement(ModAchievements.achievementMine25Diamonds);
+                break;
+            case TAG_BUILDER_HUTS:
+                this.triggerAchievement(ModAchievements.achievementBuild25Huts);
+                break;
+            case TAG_FISHERMAN_FISH:
+                this.triggerAchievement(ModAchievements.achievementCatch25Fish);
+                break;
+            case TAG_FARMER_WHEAT:
+                this.triggerAchievement(ModAchievements.achievementHarvest25Wheat);
+                break;
+            case TAG_FARMER_POTATOES:
+                this.triggerAchievement(ModAchievements.achievementHarvest25Potatoes);
+                break;
+            case TAG_FARMER_CARROTS:
+                this.triggerAchievement(ModAchievements.achievementHarvest25Carrots);
+                break;
+            case TAG_LUMBERJACK_SAPLINGS:
+                this.triggerAchievement(ModAchievements.achievementPlant25Saplings);
+                break;
+            case TAG_LUMBERJACK_TREES:
+                this.triggerAchievement(ModAchievements.achievementFell25Trees);
+                break;
+            default:
+                break;
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FOURTH)
+    }
+    /**
+     * Trigger third achievement.
+     * @param statistic the statistic.
+     */
+    private void triggerThirdAchievement(@NotNull String statistic)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "triggerThirdAchievement" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)